api-compose-object.go

```diff
@@ -315,6 +315,13 @@ func (c Client) ComposeObject(dst DestinationInfo, srcs []SourceInfo) error {
 			return fmt.Errorf("Could not get source props for %s/%s: %v", src.bucket, src.object, err)
 		}
 
+		// Error out if client side encryption is used in this source object when
+		// more than one source objects are given.
+		if i > 0 && src.Headers.Get("x-amz-meta-x-amz-key") != "" {
```

Comment:
This check won't catch the case where the first source object is encrypted while others are not. I.e, `i == 0 && src.Headers.Get("x-amz-meta-x-amz-key") != ""` is true.